tools/FeynOpt.hs

```diff
@@ -213,38 +263,51 @@ printHelp = mapM_ putStrLn lines
-  "-O3"          -> parseArgs (o3 ++ passes) verify pureCircuit xs
-  "-O4"          -> parseArgs (o4 ++ passes) verify pureCircuit xs
-  "-verify"      -> parseArgs passes True pureCircuit xs
-  "-benchmarks"  -> benchmarkFolder (head xs) >>= runBenchmarks (benchPass passes) (benchVerif verify)
+  "-purecircuit" -> parseArgs doneSwitches options {pureCircuit = True} xs
+  "-inline"      -> parseArgs doneSwitches options {passes = Inline:passes options} xs
+  "-mctExpand"   -> parseArgs doneSwitches options {passes = MCT:passes options} xs
+  "-toCliffordT" -> parseArgs doneSwitches options {passes = CT:passes options} xs
+  "-simplify"    -> parseArgs doneSwitches options {passes = Simplify:passes options} xs
+  "-phasefold"   -> parseArgs doneSwitches options {passes = Phasefold:passes options} xs
+  "-statefold"   -> parseArgs doneSwitches options {passes = (Statefold $ read (head xs)):passes options} (tail xs)
+  "-cnotmin"     -> parseArgs doneSwitches options {passes = CNOTMin:passes options} xs
+  "-tpar"        -> parseArgs doneSwitches options {passes = TPar:passes options} xs
+  "-clifford"    -> parseArgs doneSwitches options {passes = Cliff:passes options} xs
+  "-cxcz"        -> parseArgs doneSwitches options {passes = CZ:passes options} xs
+  "-decompile"   -> parseArgs doneSwitches options {passes = Decompile:passes options} xs
+  "-O2"          -> parseArgs doneSwitches options {passes = o2 ++ passes options} xs
+  "-O3"          -> parseArgs doneSwitches options {passes = o3 ++ passes options} xs
+  "-O4"          -> parseArgs doneSwitches options {passes = o4 ++ passes options} xs
+  "-verify"      -> parseArgs doneSwitches options {verify = True} xs
+  "-benchmarks"  -> benchmarkFolder (head xs) >>= runBenchmarks (benchPass $ passes options) (benchVerif $ verify options)
+  "-qasm3"       -> parseArgs doneSwitches options {useQASM3 = True} xs
+  "--"           -> parseArgs True options xs
   "VerBench"     -> runBenchmarks (benchPass [CNOTMin,Simplify]) (benchVerif True) benchmarksMedium
 --  "VerAlg"       -> runVerSuite
-  "Small"        -> runBenchmarks (benchPass passes) (benchVerif verify) benchmarksSmall
-  "Med"          -> runBenchmarks (benchPass passes) (benchVerif verify) benchmarksMedium
-  "All"          -> runBenchmarks (benchPass passes) (benchVerif verify) benchmarksAll
-  f | (drop (length f - 3) f) == ".qc" -> B.readFile f >>= runDotQC passes verify f
-  f | (drop (length f - 5) f) == ".qasm" -> readFile f >>= runQASM passes verify pureCircuit f
+  "Small"        -> runBenchmarks (benchPass $ passes options) (benchVerif $ verify options) benchmarksSmall
+  "Med"          -> runBenchmarks (benchPass $ passes options) (benchVerif $ verify options) benchmarksMedium
+  "All"          -> runBenchmarks (benchPass $ passes options) (benchVerif $ verify options) benchmarksAll
+  f | ((drop (length f - 3) f) == ".qc") || ((drop (length f - 5) f) == ".qasm") -> runFile f
```

Comment:
The extension check ends up duplicated because we want different error messages if we're "doneSwitches" or not and a filename with an unrecognizable extension is given to us -- if that filename appears before the "--", maybe it was a mistyped option; if it's after the "--" we want the user to know that we're definitely trying to interpret it as a filename, otherwise we will be misleading them.